### DISCUSSION TITLE
Reconsider configuration folder structure

### DISCUSSION THREAD

**Discussion by Samq64** [pop-os/cosmic-epoch#1245] [Ideas]
I know it's too late in the development cycle an things will likely never change but why is the configuration file structure the way it is? I understand the v1 subfolders but why prefix everything with `com.system76` and have so many single line files with some of them just being states?

I would prefer something like below and moving state-only files to `.local/state`:
```
cosmic
├── applets/
│   ├── audio.conf
│   ├── time.conf
├── AppLibrary.conf
├── CosmicBackground.conf
├── CosmicComp.conf
├── CosmicFiles/
│   ├── desktop.conf
│   ├── favorites.conf
│   └── tab.conf
├─ CosmicPanel/
|   ├── entries.conf
|   ├── panel.conf
│   └── dock.conf
├─ portal
│   └── screenshot.conf
├── CosmicSettings
|       ├── shortcuts
|       |      └── custom.conf
|       └── wallpaper.conf
├── CosmicThem
|       ├── Mode.conf
|       ├── Dark.conf
|       └── Light.conf
└── CosmicTk.conf
```

**Comment by mmstick** [comment]
> why prefix everything with `com.system76`

This is how AppIDs are written, with reverse domain name notation. GNOME uses `org.gnome.APP`, Mozilla uses `org.mozilla.APP`, and KDE uses `org.kde.APP`. Our domain is `com.system76`.

> [why] have so many single line files

Applications subscribe to the specific fields that they're interested in, either directly with an inotify watcher, or through cosmic-settings-daemon. cosmic-settings-daemon uses inotify to watch for changes to cosmic-config files, and then applications that subscribe to those fields get notified to reload those values.

> some of them just being states

States are stored in `~/.local/state/cosmic/`. It is watched and structured the same as `~/.config/cosmic/`.
